tracing-subscriber/src/registry/sharded.rs

```diff
@@ -25,22 +25,64 @@ cfg_feature!("registry", {
     /// A `Registry` is a [`Collect`] around which multiple subscribers
     /// implementing various behaviors may be [added]. Unlike other types
     /// implementing `Collect`, `Registry` does not actually record traces itself:
-    /// instead, it collects and stores span data that is exposed to any `Subscriber`s
+    /// instead, it collects and stores span data that is exposed to any [subscriber]s
     /// wrapping it through implementations of the [`LookupSpan`] trait.
     /// The `Registry` is responsible for storing span metadata, recording
     /// relationships between spans, and tracking which spans are active and which
-    /// are closed. In addition, it provides a mechanism for `Subscriber`s to store
+    /// are closed. In addition, it provides a mechanism for [subscriber]s to store
     /// user-defined per-span data, called [extensions], in the registry. This
     /// allows `Subscriber`-specific data to benefit from the `Registry`'s
     /// high-performance concurrent storage.
     ///
     /// This registry is implemented using a [lock-free sharded slab][slab], and is
     /// highly optimized for concurrent access.
     ///
+    /// # Span ID Generation
+    ///
+    /// One of the primary responsibilities of the registry is to generate [span
+    /// IDs]. Therefore, it's important for other code that interacts with the
+    /// registry, such as [subscriber]s, to understand the guarantees of the
+    /// span IDs that are generated.
```

Comment:
I'd have a one-liner at the start like...
```suggestion
    /// # Span ID Generation
    ///
    /// Span IDs are not globally unique, but the registry ensures that
    /// no two currently active spans have the same ID within a process.
    ///
    /// One of the primary responsibilities of the registry is to generate [span
    /// IDs]. Therefore, it's important for other code that interacts with the
    /// registry, such as [subscriber]s, to understand the guarantees of the
    /// span IDs that are generated.
```

...and then use the following paragraphs to explain the (relatively) dense sentence I suggested.